### DISCUSSION TITLE
v2.0.0 Technical Preview 3

### DISCUSSION THREAD

**Discussion by the1812** [the1812/Bilibili-Evolved#2027] [分享 / Share]
- 设置面板更新:
  - 修复高度过小时侧栏图标溢出面板
  - 支持批量安装功能
  - 支持设置导入/导出 (在关于面板中, 与 v1 不同的是导入后会自动刷新页面)
- 组件迁移:
  - 展开动态内容
  - 简化直播间
  - 直播勋章快速更换
  - 直播看板娘高DPI适配
  - 自动收起直播侧栏
  - 倍速记忆
  - 删除视频弹窗
  - 展开视频简介
  - 外置稍后再看
  - 启用快速收藏
  - 快捷键扩展
- 插件新增:
  - 下载视频 - aria2 输出支持
- 修复无法更新现有样式.
- 引入了 CSS `gap` 属性, 浏览器版本要求提高至 Chrome 84 / Firefox 80 / Safari 14.1
- 主脚本名称变动 (`Bilibili Evolved II`变为`Bilibili Evolved (v2)`), 更新时请多加注意.
- 生成了[功能列表文档](https://github.com/the1812/Bilibili-Evolved/blob/v2/doc/features/features.md).
> 好像只能装 GitHub 源的, jsDelivr 不知道为啥 Failed to fetch 了. 另外目前只能用 Stable 的版本, Preview 分支上还没有 v2 的文件.

相较于 v1, `快捷键扩展`功能在 v2 中支持了图形化界面设置, 并可以设置自定义按键和通过插件添加更多动作. (上面的`倍速记忆`, `外置稍后再看`, `启用快速收藏`均添加了对应动作)

<img src="https://user-images.githubusercontent.com/26504152/125065273-361d7e80-e0e4-11eb-90bb-0b145fdd5345.png" width="500">

<hr /><em>This discussion was created from the release <a href='https://github.com/the1812/Bilibili-Evolved/releases/tag/v2.0.0-tp3'>v2.0.0 Technical Preview 3</a>.</em>

**Comment by the1812** [comment]
`倍速记忆`相关功能与 v1 的对应关系:

`const rememberVideoSpeed = getComponentSettings('rememberVideoSpeed')`

| 功能             | v1                                | v2                                                  |
| ---------------- | --------------------------------- | --------------------------------------------------- |
| 记忆视频倍速     | `settings.useDefaultVideoSpeed`   | `rememberVideoSpeed.enabled`                        |
| 扩展倍速菜单     | `settings.extendVideoSpeed`       | `rememberVideoSpeed.options.extend`                 |
| 扩展倍速列表     | `settings.extendVideoSpeedList`   | `rememberVideoSpeed.options.extendList`             |
| 各视频分别记忆   | `settings.rememberVideoSpeed`     | `rememberVideoSpeed.options.individualRemember`     |
| 分别记忆倍速列表 | `settings.rememberVideoSpeedList` | `rememberVideoSpeed.options.individualRememberList` |

@LonelySteve  我将原代码中 v1 的 API 更换为了 v2, 添加了快捷键支持, 以及把文本`倍数`替换为了`倍速` (`倍速`是官方播放器的写法). 有兴趣的话可以重构下 `VideoSpeedController`, 特别是之前因 v1 架构限制可能没能实现的想法.

**Reply by LonelySteve** [reply]
我觉得【记忆视频倍速】和【扩展倍速菜单】这两个功能应该分开，若作为一个组件并且统称【倍数记忆】，感觉有点不合适。

VideoSpeedController 会重构的，现在这个样子其实过于耦合了，而且现在回过头来看，这种缓存设计也不需要了，有很大的改进空间。

**Reply by the1812** [reply]
我也觉得, 只是逻辑都在 Controller 里就先写一起了, 统称的话叫`视频倍速增强`可能合适些, 当然能分开更好

**Comment by kethoven** [comment]
如何安装啊？

**Reply by the1812** [reply]
#1939

**Comment by the1812** [comment]
第三批功能迁移计划:

- [x] 视频
  - [x] 自动展开弹幕列表
  - [x] BiliPlus跳转支持  
  - [x] 强制保留弹幕栏
  - [x] 隐藏推荐直播 / 视频推荐
- [x] 直播
  - [x] 直播间自动抽奖 _(现在还有那种抽奖弹窗吗? 我好像很久没见到过了)_
  - [x] 直播首页静音 / 隐藏推荐直播
  - [x] 直播全屏弹幕栏 / 包裹
- [x] 动态
  - [x] 复制动态/评论链接
  - [x] 动态/专栏图片导出
- [x] 启用细滚动条
- [x] 高分辨率图片
- [x] 简化首页

**Comment by CKylinMC** [comment]
关于无法直接使用jsdelivr链接安装我有一个猜测

jsdelivr有一个功能是指向根版本号时默认读取根版本号最新版本，

比如`@3.2`是最新版本的话 `@3`也可以指向这个版本。

观察报错：

![image](https://user-images.githubusercontent.com/11648014/125151130-81cd3800-e177-11eb-99cf-0d9a1bd9cc1c.png)

`@v2`被指向`@2`

会不会是这个功能导致直接`@v2`出问题？

![image](https://user-images.githubusercontent.com/11648014/125151007-ad035780-e176-11eb-96ab-6dfa4ca9be6b.png)

我尝试使用`https://cdn.jsdelivr.net/gh/the1812/Bilibili-Evolved@v2.0.0-tp3/`和`https://cdn.jsdelivr.net/gh/the1812/Bilibili-Evolved@2.0.0-tp3/`链接都可以正确打开目录。(`v2.0.0...`和`2.0.0...`)

**Reply by the1812** [reply]
不知道它为什么要把 `v2` 的 `v` 去掉, 难道前面带个 `v` 也会被识别到版本号中...
以后合到 `master` 上应该就没问题了

(甚至官网的链接转换器得到的也是 `@v2` 并且打不开)
![image](https://user-images.githubusercontent.com/26504152/125151447-c0afbd80-e178-11eb-89b0-864121433993.png)

**Reply by CKylinMC** [reply]
这个我觉得可能需要给jsdelivr反馈一下吧...虽然`@vv2`可以得到`@v2`的指向(更加迷惑了)，但我还是打不开...
